tcp-server.js

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const net = require('net');
+
+const PORT = 8000;
+
+const server = net.createServer(socket => {
+    console.log('Client connected');
+    socket.on('data', data => {
+        const request = data.toString();
+        console.log('Received request:', request);
+        console.log('Sending data')
+        socket.write('HTTP/1.1 200 OK\r\n')
+        socket.write('Content-Type: text/html\r\n')
+        socket.write('\r\n') // this line break is crucial 
+        // it lets the client know that the content is about to arrive.
+        socket.write('<h1>Big things coming tomorrow</h1>')
+        
```

Comment:
I'm really curious about how chrome can handle hanging sockets but curl and safari can't. Is this documented somewhere?

Reply:
There's a bit of browser hackery happening here. You'll find if you increase the amount of HTML you output, different browsers will respond differently. Maybe even throwing in </html> you'll get a different behaviour. It's all because browsers were trying different hacks back in the days before we had websockets. You can read a bit more about it here https://en.wikipedia.org/wiki/Comet_(programming)